examples/platform/linux/Options.cpp

```diff
@@ -477,7 +477,7 @@ bool HandleOption(const char * aProgram, OptionSet * aOptions, int aIdentifier,
     }
 
     case kDeviceOption_Passcode:
-        LinuxDeviceOptions::GetInstance().payload.setUpPINCode = static_cast<uint32_t>(strtoul(aValue, nullptr, 0));
+        LinuxDeviceOptions::GetInstance().payload.setUpPINCode = static_cast<uint32_t>(atoi(aValue));
```

Comment:
Pending for @arkq for review so we do not bounce back and forth here. Maybe we should check for 0 prefix and assume that is decimal or have other heuristics (like require 0x or `hex:` prefix for hex). I am not sure what kind of passcodes people paste.

Reply:
@greenstatic I see that #34172 mentions the 0x Prefix for hex. How about checking aValue prefix and if it starts with 0x then you use strtoul and otherwise atoi?

Reply:
That sounds like a nice compromise. Maybe wouldn't be a bad idea to introduce this to parsing also the other fields the aforementioned PR changed. @andy31415 What do you think?

Reply:
Some reusable "parse integer" would be nice as well for sure.